scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,255 @@
+                 "auth_domain_email", "auth_domain_creation_error",
+                 "workspace_link", "workspace_creation_error",
+                 "workspace_ACLs", "workspace_ACLs_error",
+                 "workspace_setup_status"]
+    output_df = pd.DataFrame(columns=col_names)
+
+    # per row in tsv/df
+    for row in setup_info_df.index:
+        # authorization domain
+        auth_domain_name = setup_info_df.loc[row].get(key='auth_domain_name')
+        output_df.at[row, "input_auth_domain_name"] = auth_domain_name
+        auth_status = create_auth_domain(auth_domain_name)
+
+        if auth_status != 201:  # if auth creation fail
+            output_df.at[row, "auth_domain_creation_error"] = auth_status  # write error response to col
+            continue  # go to next row in df
+        else:  # if auth creation success
+            # TODO: will i need auth_domain_email again?
+            auth_domain_email = f"{auth_domain_name}@firecloud.org"
+            output_df.at[row, "auth_domain_email"] = auth_domain_email
+
+        # workspace
+        workspace_name = setup_info_df.loc[row].get(key='workspace_name')
+        output_df.at[row, "input_workspace_name"] = workspace_name
+        create_ws_request = make_create_workspace_request(workspace_name, auth_domain_name, project)  # get json for createWorkspace
+
+        create_workspace_status = create_workspace(create_ws_request, workspace_name, project)  # createWorkspace
+
+        # created or user wants to continue with existing
+        if create_workspace_status in [201, "Y"]:
+            output_df.at[row, "workspace_link"] = f"https://app.terra.bio/#workspaces/{project}/{workspace_name}"
+            # add member ACLs to workspace
+            member_status_code, member_response = add_members_to_workspace(workspace_name, auth_domain_name, project)
+            if member_status_code == 200:
+                output_df.at[row, "workspace_ACLs"] = member_response  # add emails to df
+                output_df.at[row, "workspace_setup_status"] = "Success"  # denotes final workspace setup step
+            else:
+                output_df.at[row, "workspace_ACLs_error"] = member_response
+                output_df.at[row, "workspace_setup_status"] = "Failed"
+        # if "N" (user does not want to continue with existing workspace) or other error
+        else:
+            output_df.at[row, "workspace_creation_error"] = create_workspace_status  # write error response to col
+            output_df.at[row, "workspace_ACLs_error"] = "Incomplete"  # adding members function not called
+            output_df.at[row, "workspace_setup_status"] = "Failed"
+            continue
+
+    timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    output_df.to_csv(f"{timestamp}_workspaces_setup_status.tsv", sep="\t", index=False)
+
+    print("If applicable, workspace creation and set-up complete. Check output file for full details.")
```

Comment:
could you count the number of successes and report that? like `10/10 workspaces created and set up successfully` ? so that if/when it all does go well you don't have to look at the setup tsv?